subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNodeExecutor.java

```diff
@@ -127,19 +129,30 @@ public void visitFileTreeBackedByFile(File file, FileTreeInternal fileTree, File
             });
         inputHierarchy.recordNodeAccessingLocations(node, taskInputs);
         for (String locationConsumedByThisTask : taskInputs) {
-            outputHierarchy.getNodesAccessing(locationConsumedByThisTask).stream()
-                .filter(producerNode -> hasNoSpecifiedOrder(producerNode, node))
-                .forEach(producerWithoutDependency -> collectValidationProblem(producerWithoutDependency, node, validationContext));
+            collectValidationProblemsForConsumer(node, validationContext, outputHierarchy.getNodesAccessing(locationConsumedByThisTask));
         }
         for (FilteredTree filteredFileTreeInput : filteredFileTreeTaskInputs) {
             Spec<FileTreeElement> spec = filteredFileTreeInput.getPatterns().getAsSpec();
             inputHierarchy.recordNodeAccessingFileTree(node, filteredFileTreeInput.getRoot(), spec);
-            outputHierarchy.getNodesAccessing(filteredFileTreeInput.getRoot(), spec).stream()
-                .filter(producerNode -> hasNoSpecifiedOrder(producerNode, node))
-                .forEach(producerWithoutDependency -> collectValidationProblem(producerWithoutDependency, node, validationContext));
+            collectValidationProblemsForConsumer(node, validationContext, outputHierarchy.getNodesAccessing(filteredFileTreeInput.getRoot(), spec));
         }
     }
 
+    private void collectValidationProblemsForConsumer(LocalTaskNode consumer, TypeValidationContext validationContext, Collection<Node> producers) {
+        producers.stream()
+            .filter(producerNode -> hasNoSpecifiedOrder(producerNode, consumer))
+            .filter(LocalTaskNodeExecutor::isEnabled)
+            .forEach(producerWithoutDependency -> collectValidationProblem(producerWithoutDependency, consumer, validationContext));
+    }
+
+    private static boolean isEnabled(Node node) {
```

Comment:
Would it make sense to have this directly on `LocalTaskNode`?

Reply:
I don't think so. You shouldn't call this method on the node when you build the execution graph, so I wouldn't move this there.